rpcserver.go

```diff
@@ -5874,8 +5878,20 @@ func (r *rpcServer) ProveAssetOwnership(ctx context.Context,
 		return nil, fmt.Errorf("error fetching commitment: %w", err)
 	}
 
+	var challengeOpt fn.Option[[32]byte]
+
+	switch len(req.Challenge) {
+	case 0:
+		challengeOpt = fn.None[[32]byte]()
```

Comment:
No need to set `fn.None()`, that's the default value of a `var ... fn.Option[...]`.